### ISSUE TITLE
SpectralAngleMapper metric nan when any pixel is zero

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#1047]
### Environment

all

### Describe the bug

Duplicate report of bug https://github.com/Lightning-AI/torchmetrics/issues/3322 affecting `deepinv.metric.SpectralAngleMapper`

### Steps to reproduce

```python
a, b = torch.ones(2, 1, 3, 8, 8)
a[:, :, 5, 3] = 0
dinv.metric.SpectralAngleMapper()(a, b)

out: tensor([nan])
```

### Expected behavior

The NaN values are ignored during reduction.

### Fix

Luckily in deepinv we do our own reduction. So we could just reduce ignoring nans here (`mean` -> `nanmean`):

https://github.com/deepinv/deepinv/blob/c40dd4c9fa2704c18bae7bb9109517a1c03c228e/deepinv/loss/metric/distortion.py#L590-L592
